backends/dpdk/dpdkArch.cpp

```diff
@@ -1665,22 +1662,56 @@ const IR::Node *CopyMatchKeysToSingleStruct::postorder(IR::KeyElement *element)
         insertions = it->second;
     }
 
-    auto keyName = element->expression->toString();
+    cstring keyName;
+    if (auto pe = element->expression->to<IR::PathExpression>()) {
+        keyName = pe->path->name;
+    } else if (auto mem = element->expression->to<IR::Member>()) {
+        /* We expect all nested structures are flattened and only
+         * expected expressions are 'h.<hdrname>.<fieldname> or m.<fieldname> */
```

Comment:
Must every header really start with `h`? This seems like an odd restriction. If this restriction is not there you can likely resolve everything recursively.

Reply:
Yes. Dpdk backend converts all headers into this format in the process of moving closer to the target spec file format.

Reply:
Ah, that enforcement looks good to me then! This could also be a helper which you could use in other parts of the code for validation, if necessary.

Reply:
Added a helper